core/trino-spi/src/main/java/io/trino/spi/Page.java

```diff
@@ -315,6 +316,21 @@ public Page getLoadedPage(int... columns)
         return wrapBlocksWithoutCopy(positionCount, blocks);
     }
 
+    public Page getLoadedPage(Predicate<Integer> loadChannel, int... columns)
```

Comment:
it would be clearer if it's `getLoadedPage(int[] columns, int[] eagerlyLoadedColumns)`

Reply:
Alternatively, you could keep the method local to `InputPages`